demos/source/aws_iot_demo_jobs.c

```diff
@@ -0,0 +1,533 @@
+                                       JOB_ACTION_KEY_LENGTH,
+                                       action,
+                                       actionLength );
+}
+
+/**
+ * @brief Get and update the job.
+ */
+static int32_t _executeDemo( IotMqttConnection_t const mqttConnection,
+                             const char * pThingName,
+                             size_t thingNameLength )
+{
+    bool success = false;
+    AwsIotJobsError_t err = AWS_IOT_JOBS_SUCCESS;
+    AwsIotJobState_t result = AWS_IOT_JOB_STATE_FAILED;
+    AwsIotJobsResponse_t resp = AWS_IOT_JOBS_RESPONSE_INITIALIZER;
+    AwsIotJobsRequestInfo_t req = AWS_IOT_JOBS_REQUEST_INFO_INITIALIZER;
+    AwsIotJobsUpdateInfo_t updateInfo = AWS_IOT_JOBS_UPDATE_INFO_INITIALIZER;
+    const char * pAction = NULL;
+    size_t actionLength = 0;
+    const char * pMessage = NULL;
+    size_t messageLength = 0;
+
+    req.mqttConnection = mqttConnection;
+    req.pThingName = pThingName;
+    req.thingNameLength = thingNameLength;
+    req.pJobId = _pJobId + 1;
+    req.jobIdLength = _JobIdLength - 2;
+    req.mallocResponse = malloc;
+
+    /* Check that a job notification came in. */
+    if( _pJobId[ 0 ] != '\0' )
+    {
+        /* Update Job to IN_PENDING. */
+        updateInfo.newStatus = AWS_IOT_JOB_STATE_IN_PROGRESS;
+
+        err = AwsIotJobs_UpdateSync(
+            &req, &updateInfo, 0, AWS_IOT_DEMO_JOBS_MQTT_TIMEOUT, &resp );
+
+
+        if( err != AWS_IOT_JOBS_SUCCESS )
+        {
+            IotLogError( "Update Failed: %s", AwsIotJobs_strerror( err ) );
+            success = false;
+        }
+	else
+	{
+	    success = true;
+	}
+    }
+
+    /* Parse out the action. */
+    if( success )
+    {
+        success = _getAction( &pAction, &actionLength );
+        if( !success )
+        {
+            IotLogError( "Failed to find %s Key.", JOB_ACTION_KEY );
+        }
+    }
+
+    /* Process the action. */
+    if( success )
+    {
+        /* We are only going to look at the first char after the '"'.
+	   This is just to simplify the demo. */
+        switch( pAction[ 1 ] )
+        {
+        case 'p':
+            if( _getMessage( &pMessage, &messageLength ) )
+            {
+                IotLogInfo( "Executing Print: %.*s", messageLength, pMessage );
+                result = AWS_IOT_JOB_STATE_SUCCEEDED;
+            }
+            else
+            {
+                IotLogInfo( "Failed to execute print." );
+                result = AWS_IOT_JOB_STATE_FAILED;
+            }
+            break;
+        case 'w':
+            result = AWS_IOT_JOB_STATE_SUCCEEDED;
+            IotClock_SleepMs( 10000 );
+            /*
+	    if ( _getMessage( &pMessage, &messageLength ) )
+	    {
+		IotLogInfo( "" );
+		result = AWS_IOT_JOB_STATE_SUCCEEDED;
+	    }
+            else
+            {
+                IotLogInfo( "Failed to execute wait." );
+                result = AWS_IOT_JOB_STATE_FAILED;
+            }
+	    */
+            break;
+        default:
```

Comment:
Maybe add a "publish" action.  For example:

`{"action":"publish","message":"Hello world!"}` sends "Hello world!" on the MQTT connection.

Reply:
What topic?

Reply:
Have the topic in the Job document.

```{"action":"publish","message":"Hello world!","topic":"demo/jobs"}```